app/services/kredits.js

```diff
@@ -33,20 +33,10 @@ export default Service.extend({
   // this is called in the routes beforeModel().  So it is initialized before everything else
   // and we can rely on the ethProvider and the potential currentUserAccounts to be available
   getEthProvider: function() {
-    return new RSVP.Promise((resolve) => {
-      let ethProvider;
-      if (typeof window.web3 !== 'undefined') {
-        console.debug('[kredits] Using user-provided instance, e.g. from Mist browser or Metamask');
-        ethProvider = new ethers.providers.Web3Provider(window.web3.currentProvider);
-        ethProvider.listAccounts().then((accounts) => {
-          this.set('currentUserAccounts', accounts);
-          const ethSigner = accounts.length === 0 ? null : ethProvider.getSigner();
-          resolve({
-            ethProvider,
-            ethSigner
-          });
-        });
-      } else {
+    let ethProvider;
+
+    return new RSVP.Promise(async (resolve) => {
+      function instantiateWithoutAccount () {
```

Comment:
do those functions need to be in that RSVP.Promise ? 
or could those be general functions that return a Promise that resolves the ethProvider? 

something like: 
```js
getEthProvider: function() {
  if (window.ethereum) {
    return window.ethereum.enable().then(() => {
      return instantiateWithAccount(...)
    })
  }
}
```

Reply:
It's easier this way. Anyone wanting to refactor it can do so in the future.